### PR TITLE
Add G Suite New User List Component

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Creates a new G Suite Component to create a list of new G Suite Users

This new Component is not used anywhere yet, this PR is just to introduce it. See #32417 for the full changes.

#### Testing instructions

1. Navigate to `/devdocs/design/g-suite-examples`
1. Test the component, including performance at small widths, performance of the form, etc.

### CODE REVIEW

File: client/components/gsuite/gsuite-new-user-list/domains-select.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+import FormSelect from 'components/forms/form-select';
+
+interface Props {
+	domains: string[];
+	onChange: ( event: ChangeEvent< HTMLInputElement > ) => void;
+	value: string;
+}
+
+const DomainsSelect = ( { domains, onChange, value }: Props ) => {
```
Comment (stephanethomas) [inline client/components/gsuite/gsuite-new-user-list/domains-select.tsx]: Shouldn't we name this component `GSuiteDomainsSelect`?

File: client/components/gsuite/gsuite-new-user-list/style.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+.gsuite-new-user-list__new-user-email {
+	display: flex;
+	flex-direction: column;
+	flex: 1;
+
+	// input {
+	// 	margin: 0 0 15px;
+	// 	width: auto;
+	// }
```
Comment (stephanethomas) [inline client/components/gsuite/gsuite-new-user-list/style.scss]: We probably want to remove those commented lines.

File: client/components/gsuite/gsuite-new-user-list/style.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		select {
+			border-bottom-left-radius: 0;
+			border-bottom-width: 1px;
+			border-top-left-radius: 0;
+			margin-left: -2px;
+		}
+	}
+}
+
+.gsuite-new-user-list__new-user-name-conatiner {
```
Comment (stephanethomas) [inline client/components/gsuite/gsuite-new-user-list/style.scss]: Typo 😉.

File: client/lib/gsuite/new-users.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	value,
+	error:
+		! error && ( ! value || '' === value ) ? i18n.translate( 'This field is required.' ) : error,
+} );
+
+const sixtyCharacterField = ( { value, error }: GSuiteNewUserField ): GSuiteNewUserField => ( {
+	value,
+	error:
+		! error && 60 < value.length
+			? i18n.translate( "This field can't be longer than 60 characters." )
```
Comment (stephanethomas) [inline client/lib/gsuite/new-users.ts]: We could extract the number of characters to be able to reuse this string in a different context where another limit would apply.

File: client/components/gsuite/docs/new-user-list.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		<Card>
+			<GSuiteNewUserList
+				domains={ domains }
+				extraValidation={ useExtraValidation ? extraValidation : user => user }
+				selectedDomainName={ domainOne.name }
+				onUsersChange={ changedUsers => setUsers( changedUsers ) }
+				users={ users }
+			>
+				{ 0 < users.length && users.every( userIsReady ) ? (
+					<span>{ '✅ - All Users Ready' }</span>
```
[the remark this answers is not in the document]
Comment (belcherj) [reply client/components/gsuite/docs/new-user-list.tsx]: fixed

File: client/components/gsuite/gsuite-new-user-list/domains-select.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	return (
+		<FormSelect
+			className="gsuite-new-user-list__new-user-domain-select"
```
[the remark this answers is not in the document]
Comment (belcherj) [reply client/components/gsuite/gsuite-new-user-list/domains-select.tsx]: fixed

File: client/components/gsuite/gsuite-new-user-list/domains-select.tsx (hunk 1 above)
Comment (belcherj) [reply client/components/gsuite/gsuite-new-user-list/domains-select.tsx]: fixed

File: client/components/gsuite/gsuite-new-user-list/style.scss (hunk 2 above)
Comment (belcherj) [reply client/components/gsuite/gsuite-new-user-list/style.scss]: fixed

File: client/components/gsuite/gsuite-new-user-list/style.scss (hunk 3 above)
Comment (belcherj) [reply client/components/gsuite/gsuite-new-user-list/style.scss]: fixed

File: client/lib/gsuite/new-users.ts (hunk 4 above)
Comment (belcherj) [reply client/lib/gsuite/new-users.ts]: fixed

**Pull request merged**